src/main/java/org/folio/rest/core/RestClient.java

```diff
@@ -102,4 +121,97 @@ public CompletableFuture<Void> handlePutRequest(String endpoint, JsonObject reco
 
     return future;
   }
+
+  public  CompletableFuture<JsonObject> post(String endpoint, JsonObject recordData) {
+    CompletableFuture<JsonObject> future = new FolioVertxCompletableFuture<>(ctx);
+
+    if (logger.isDebugEnabled()) {
+      logger.debug("Sending 'POST {}' with body: {}", endpoint, recordData.encodePrettily());
+    }
+
+    try {
+      httpClient
+        .request(HttpMethod.POST, recordData.toBuffer(), endpoint, okapiHeaders)
+        .thenApply(HelperUtils::verifyAndExtractBody)
+        .handle((body, t) -> {
+          if (t != null) {
+            logger.error("'POST {}' request failed with error {}. Request body: {}", t.getCause(), endpoint, recordData.encodePrettily());
+            future.completeExceptionally(t.getCause());
+          } else {
+            if (logger.isDebugEnabled()) {
+              logger.debug("'POST {}' request successfully processed. Record with '{}' id has been created", endpoint, body);
+            }
+            future.complete(body);
+          }
+          return null;
+        });
+    } catch (Exception e) {
+      logger.error("'POST {}' request failed with error {}. Request body: {}", e, endpoint, recordData.encodePrettily());
+      future.completeExceptionally(e);
+    }
+
+    return future;
+  }
+
+  public CompletableFuture<Void> delete(String endpointById) {
+    CompletableFuture<Void> future = new FolioVertxCompletableFuture<>(ctx);
+    if (logger.isDebugEnabled()) {
```

Comment:
You won't get any performance improvement because you're passing data already computed. You can remove isDebugEnabled and configure via file

Reply:
Aggree. In this case logger parameters do not have computations or encodings, so it will not affect performance. fixed